dependencies/ih264d/CMakeLists.txt

```diff
@@ -134,10 +115,71 @@ add_library (ih264d
 "decoder/ih264d_vui.h"
 "decoder/iv.h"
 "decoder/ivd.h"
+)
+
+if (CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "amd64" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "AMD64")
+set(LIBAVCDEC_X86_INCLUDES "common/x86" "decoder/x86")
+include_directories("common/" "decoder/" ${LIBAVCDEC_X86_INCLUDES})
+target_sources(ih264d PRIVATE
+"common/x86/ih264_chroma_intra_pred_filters_ssse3.c"
+"common/x86/ih264_deblk_chroma_ssse3.c"
+"common/x86/ih264_deblk_luma_ssse3.c"
+"common/x86/ih264_ihadamard_scaling_sse42.c"
+"common/x86/ih264_ihadamard_scaling_ssse3.c"
+"common/x86/ih264_inter_pred_filters_ssse3.c"
+"common/x86/ih264_iquant_itrans_recon_dc_ssse3.c"
+"common/x86/ih264_iquant_itrans_recon_sse42.c"
+"common/x86/ih264_iquant_itrans_recon_ssse3.c"
+"common/x86/ih264_luma_intra_pred_filters_ssse3.c"
+"common/x86/ih264_mem_fns_ssse3.c"
+"common/x86/ih264_padding_ssse3.c"
+"common/x86/ih264_platform_macros.h"
+"common/x86/ih264_resi_trans_quant_sse42.c"
+"common/x86/ih264_weighted_pred_sse42.c"
 "decoder/x86/ih264d_function_selector.c"
 "decoder/x86/ih264d_function_selector_sse42.c"
 "decoder/x86/ih264d_function_selector_ssse3.c"
 )
+elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
```

Comment:
This should (probably) be 

```elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")```

for compilation of this submodule on macos, as it reports the CPU architecture as arm64.

Reply:
In addition, ih264 throws a lot of errors when trying to compile on macos. 

```/var/folders/gg/cvxn27gj4px87t7ysvyycdzr0000gn/T/ih264_intra_pred_chroma_av8-9d53ce.s:180:5: error: ADR/ADRP relocations must be GOT relative```